libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts

```diff
@@ -18,22 +18,34 @@ import { ComponentDeclarativeTypes } from '../componentDeclarativeTypes';
 import { DeclarativeType } from '../declarativeType';
 import { CustomDeserializer } from '../customDeserializer';
 import { DefaultLoader } from '../defaultLoader';
+import { ResourceExplorerOptions } from './resourceExplorerOptions';
 
 /**
  * Class which gives standard access to content resources.
  */
 export class ResourceExplorer {
+    private _declarativeTypes: ComponentDeclarativeTypes[];
     private _kindToType: Map<string, new () => unknown> = new Map();
     private _kindDeserializer: Map<string, CustomDeserializer<unknown, unknown>> = new Map();
     private _eventEmitter: EventEmitter = new EventEmitter();
     private _typesLoaded = false;
 
     /**
      * Initializes a new instance of the `ResourceExplorer` class.
-     * @param providers Resource providers.
+     *
+     * @param {ResourceProvider[] | ResourceExplorerOptions} providersOrOptions Resource providers or resource explorer options.
      */
-    public constructor(providers: ResourceProvider[] = []) {
-        this.resourceProviders = providers;
+    public constructor(providersOrOptions: ResourceProvider[] | ResourceExplorerOptions = []) {
```

Comment:
I've started a new pattern for public overloads that I think is handy with respect to external developer experience.

```typescript
/**
 * JSDoc for this overload
 */
public constructor(providers: ResourceProvider[]);
/**
 * JSDoc for this overload
 */
public constructor(options: ResourceExplorerOptions);
public constructor(providersOrOptions: ResourceProvider[] | ResourceExplorerOptions) {
  // logic
}
```

The final overload does not include a JSDoc comment, so only the first two show up in vscode (or other tools). To me, this makes it very clear that each constructor is different.